pkg/machines/libvirt-dbus.js

```diff
@@ -379,11 +379,16 @@ LIBVIRT_DBUS_PROVIDER = {
                     });
         }
 
-        if (options.destroy) {
-            return destroy().then(undefine());
-        } else {
-            return undefine();
-        }
+        return call(connectionName, objPath, "org.freedesktop.DBus.Properties", "Get", ["org.libvirt.Domain", "Persistent"], TIMEOUT)
+                .then(persistent => {
+                    if (options.destroy)
+                        if (!persistent[0].v)
+                            return destroy();
+                        else
+                            return destroy().then(undefine());
+                    else
+                        return undefine();
```

Comment:
This LBYL pattern feels a bit clumsy with the extra D-Bus call. Would an EAFP approach be possible here? I. e. always undefine, and ignore if it fails with "this domain does not exist"?

Reply:
Done